### ISSUE TITLE
calculation of loss_gt 

### ISSUE THREAD

**Issue by THBUer-yw** [thomasgauthier/LLM-self-play#2]
In the line 36 of the .py file, should it be **torch.log(model_probs_gt / opponent_probs_gt)**, without the minus?

**Comment by thomasgauthier** [comment]
Yes the minus shouldn't have been there. There was some confusion as to what we're trying to minimize/maximize.

In ce23eb5 I revisited the loss function and tried to get it more in line with the formula shown in the paper. (correctly deriving probability distributions from the logits, summing logistic loss at each training step)

**Comment by thomasgauthier** [comment]
@THBUer-yw If you are looking at this for more than educational purposes I would point you to a different implementation that seems more production-ready : https://github.com/lucidrains/self-rewarding-lm-pytorch/blob/main/self_rewarding_lm_pytorch/spin.py

**Comment by THBUer-yw** [comment]
thanks
